rules/rules_test.go

```diff
@@ -667,6 +670,60 @@ func TestExpressions(t *testing.T) {
 			},
 		},
 		{
+			expr: `http_requests{group="canary"} or http_requests{group="production"}`,
+			output: []string{
+				`http_requests{group="canary", instance="0", job="api-server"} => 300 @[%v]`,
+				`http_requests{group="canary", instance="0", job="app-server"} => 700 @[%v]`,
+				`http_requests{group="canary", instance="1", job="api-server"} => 400 @[%v]`,
+				`http_requests{group="canary", instance="1", job="app-server"} => 800 @[%v]`,
+				`http_requests{group="production", instance="0", job="api-server"} => 100 @[%v]`,
+				`http_requests{group="production", instance="0", job="app-server"} => 500 @[%v]`,
+				`http_requests{group="production", instance="1", job="api-server"} => 200 @[%v]`,
+				`http_requests{group="production", instance="1", job="app-server"} => 600 @[%v]`,
+			},
+		},
+		{
+			expr: `http_requests{group="canary"} / by(instance,job) http_requests{group="production"}`,
+			output: []string{
+				`{instance="0", job="api-server"} => 3 @[%v]`,
+				`{instance="0", job="app-server"} => 1.4 @[%v]`,
+				`{instance="1", job="api-server"} => 2 @[%v]`,
+				`{instance="1", job="app-server"} => 1.3333333333333333 @[%v]`,
+			},
+		},
+		{
+			// many-to-one
+			expr: `http_requests{group="canary"} / group_left by(instance) http_requests{group="production",job="api-server"}`,
+			output: []string{
+				`{instance="1"} => 4 @[%v]`,
```

Comment:
Unless I'm very mistaken, we should never return a vector that has multiple elements identifying the same time series. Besides that not being very sensical, that will screw up code in any places where we use the labelset as a means to compute a unique identifier of a vector element.